modules/api/src/main/java/org/apache/ignite/compute/JobDescriptor.java

```diff
@@ -33,22 +33,26 @@ public class JobDescriptor<T, R> {
 
     private final JobExecutionOptions options;
 
+    private final @Nullable Marshaller<T, byte[]> argumentMarshaller;
+
     private final @Nullable Marshaller<R, byte[]> resultMarshaller;
 
-    private final @Nullable Marshaller<T, byte[]> argumentMarshaller;
+    private final Class<?> resultClass;
```

Comment:
```suggestion
    private final Class<R> resultClass;
```

Reply:
I tried it and did not face any issues, `pojoJobWithDifferentClass` compiles and works: https://github.com/apache/ignite-3/pull/4447/commits/5debde84718c39f9fcff964552cb29d752f7b1a5